cmd/agent/DEVELOPERS.md

```diff
@@ -3,6 +3,18 @@
 This document provides relevant instructions for maintainers of the Grafana
 Agent.
 
+## Branching Strategy
+
+The Grafana Agent's branching strategy has shifted with the release of v0.19.0
+and is moving to a main always releasable approach. This will allow us to better control
+the release cadence and ensure that the project has regular releases. 
+
+For contributors, any large change should be made in a `dev.*` branch. These branches allow work
+on large features without crowding main. If you would like to work on a large feature let @rfratto or
+@mattdurham know, and we can create a branch for you.
```

Comment:
Maybe instead of our usernames, let's ask them to open a discussion. The list of core maintainers will change over time and I don't want this to get out of sync (also that way Mario or Joe can also create these branches)